include/zephyr/bluetooth/audio/ccp.h

```diff
@@ -0,0 +1,95 @@
+/**
+ * @file
+ * @brief Bluetooth Call Control Profile (CCP) APIs.
+ */
+
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_BLUETOOTH_AUDIO_CCP_H_
+#define ZEPHYR_INCLUDE_BLUETOOTH_AUDIO_CCP_H_
+
+/**
+ * @brief Call Control Profile (CCP)
+ *
+ * @defgroup bt_ccp Call Control Profile (CCP)
+ *
+ * @since 3.7
+ * @version 0.1.0
+ *
+ * @ingroup bluetooth
+ * @{
+ *
+ * Call Control Profile (CCP) provides procedures to initiate and control calls.
+ * It provides the Call Control Client and the Call Control Server roles,
+ * where the former is usually placed on resource constrained devices like headphones,
+ * and the latter placed on more powerful devices like phones and PCs.
+ *
+ * The profile is not limited to carrier phone calls and can be used with common applications like
+ * Discord and Teams.
+ */
+
+#include <zephyr/bluetooth/audio/tbs.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief Abstract Call Control Server Telephone Bearer structure. */
+struct bt_ccp_server_bearer;
+
+/**
+ * @brief Register a Telephone Bearer
+ *
+ * This will register a Telephone Bearer Service (TBS) (or a Generic Telephone Bearer service
+ * (GTBS)) with the provided parameters.
+ *
+ * As per the TBS specification, the GTBS shall be instantiated for the feature,
+ * and as such a GTBS shall always be registered before any TBS can be registered.
+ * Similarly, all TBS shall be unregistered before the GTBS can be unregistered with
+ * bt_ccp_server_unregister_bearer().
+ *
+ * @param[in]  param   The parameters to initialize the bearer.
+ * @param[out] bearer  Pointer to the initialized bearer.
+ *
+ * @retval 0 Success
+ * @retval -EINVAL @p param contains invalid data
+ * @retval -EALREADY @p param.gtbs is true and GTBS has already been registered
+ * @retval -EAGAIN @p param.gtbs is false and GTBS has not been registered
+ * @retval -ENOMEM @p param.gtbs is false and no more TBS can be registered (see
+ *         @kconfig{CONFIG_BT_TBS_BEARER_COUNT})
+ * @retval -ENOEXEC The service failed to be registered
+ */
+int bt_ccp_server_register_bearer(const struct bt_tbs_register_param *param,
+				  struct bt_ccp_server_bearer **bearer);
```

Comment:
For reviewers: I consider the following naming options:

2. bt_ccp_ccs (Bluetooth Call Control Profile Call Control Server, where Call Control Server is the actual name of the role in CCP)
3. bt_ccp_call_control_server (more verbose version of the above; naming similar to e.g. bt_bap_unicast_server)
4. bt_ccp_call_ctrl_srv (shorter version of 3, more similar to the VCP and MICP APIs (e.g. bt_vcp_vol_rend)
5. bt_ccp_server (and in upcoming PRs, bt_ccp_client). 

If you have any preferences or other suggestions, please let me know and we can just do a rename

Reply:
I prefer 3. bt_ccp_call_control_server. It is easier to spelling and understand.

Reply:
@koffes is that a vote for 3) then?

Reply:
Yes, 4. is good as well.

Reply:
prefer 3.